OpenRA.Mods.Common/MapFormats/DefaultMap.cs

```diff
@@ -0,0 +1,1451 @@
+		public Ruleset Rules { get; protected set; }
+		Ruleset IMap.Rules => Rules;
+		public CellLayer<ResourceTile> Resources { get; protected set; }
+		public CellLayer<byte> Height { get; protected set; }
+		public CellLayer<byte> Ramp { get; protected set; }
+		public CellLayer<byte> CustomTerrain { get; private set; }
+
+		public PPos[] ProjectedCells { get; private set; }
+		public CellRegion AllCells { get; private set; }
+		public List<CPos> AllEdgeCells { get; private set; }
+
+		public event Action<CPos> CellProjectionChanged;
+		Dictionary<CPos, TerrainTile> IMap.ReplacedInvalidTerrainTiles => ReplacedInvalidTerrainTiles;
+		public int2 MapSize { get; protected set; }
+		MapGrid IMap.Grid => Grid;
+		int2 IMap.MapSize => MapSize;
+		protected CVec[][] GridTilesByDistance => Grid.TilesByDistance;
+		public Rectangle Bounds;
+		Rectangle IMap.Bounds => Bounds;
+
+		// Generated data
+
+		/// <summary>Defines the order of the fields in map.yaml</summary>
+		static readonly MapField[] YamlFields =
+		{
+			new MapField("MapFormat"),
+			new MapField("RequiresMod"),
+			new MapField("Title"),
+			new MapField("Author"),
+			new MapField("Tileset"),
+			new MapField("MapSize"),
+			new MapField("Bounds"),
+			new MapField("Visibility"),
+			new MapField("Categories"),
+			new MapField("Translations", required: false, ignoreIfValue: ""),
+			new MapField("LockPreview", required: false, ignoreIfValue: "False"),
+			new MapField("Players", "PlayerDefinitions"),
+			new MapField("Actors", "ActorDefinitions"),
+			new MapField("Rules", "RuleDefinitions", required: false),
+			new MapField("Sequences", "SequenceDefinitions", required: false),
+			new MapField("ModelSequences", "ModelSequenceDefinitions", required: false),
+			new MapField("Weapons", "WeaponDefinitions", required: false),
+			new MapField("Voices", "VoiceDefinitions", required: false),
+			new MapField("Music", "MusicDefinitions", required: false),
+			new MapField("Notifications", "NotificationDefinitions", required: false),
+		};
+
+		// Format versions
+		public int MapFormat { get; private set; }
+		int IMap.MapFormat => MapFormat;
+
+		public readonly byte TileFormat = 2;
+
+		public static string ComputeUID(IReadOnlyPackage package)
+		{
+			return ComputeUID(package, GetMapFormat(package));
+		}
+
+		public static string ComputeUID(IReadOnlyPackage package, int format)
+		{
+			// UID is calculated by taking an SHA1 of the yaml and binary data
+			var requiredFiles = new[] { "map.yaml", "map.bin" };
+			var contents = package.Contents.ToList();
+			foreach (var required in requiredFiles)
+				if (!contents.Contains(required))
+					throw new FileNotFoundException("Required file {0} not present in this map".F(required));
+
+			var streams = new List<Stream>();
+			try
+			{
+				foreach (var filename in contents)
+					if (filename.EndsWith(".yaml") || filename.EndsWith(".bin") || filename.EndsWith(".lua") || (format >= 12 && filename == "map.png"))
```

Comment:
> Reviewed the first commit. Running `make` throws tons of errors, and the commit should be renamed to something like: "Abstracted map loading to DefaultMapLoader : IMapLoader"

Yes there have been numerous PRs since then which prevents this change from working in isolation. Also because of how tightly coupled the loader is with the new map format, I would have to revert the new map format changes to have this commit work in isolation. However the changes that were made are still valid and can be reviewed in isolation, provided that you still refer to the latest commit for the working version.